### PR TITLE
feat(paas_ready): get all configuration from engine

### PR DESCRIPTION
**Pull request by baptistemesta**
instead of reading it from the filesystem all configurations are get
from the engine and put in the ConfigurationFilesManager

The PlatformTenantListener initialize it with configuration returned by
the platformApi

### CODE REVIEW

File: common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFile.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            return ConfigurationFilesManager.getInstance().getTenantProperties(propertiesFilename, tenantId);
+        }
+        return ConfigurationFilesManager.getInstance().getPlatformProperties(propertiesFilename);
+    }
+
+    public void removeProperty(final String propertyName) {
+        try {
+            ConfigurationFilesManager.getInstance().removeProperty(propertiesFilename, tenantId, propertyName);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```
Comment (educhastenier) [inline common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFile.java]: What is the consequence of throwing Runtime Exception?

File: common/src/main/java/org/bonitasoft/console/common/server/utils/PlatformManagementUtils.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    private void platformLogout(PlatformSession platformSession) throws BonitaException {
+        PlatformLoginAPI platformLoginAPI = PlatformAPIAccessor.getPlatformLoginAPI();
+        platformLoginAPI.logout(platformSession);
+    }
+
+    private void retrieveTenantsConfiguration(PlatformAPI platformAPI) throws IOException {
+        Map<Long, Map<String, byte[]>> clientPlatformConfigurations = platformAPI.getClientTenantConfigurations();
+        for (Map.Entry<Long, Map<String, byte[]>> longMapEntry : clientPlatformConfigurations.entrySet()) {
```
Comment (educhastenier) [inline common/src/main/java/org/bonitasoft/console/common/server/utils/PlatformManagementUtils.java]: rename local variable

**Comment by educhastenier** [comment]
Seems that you missed to refactor **DefaultFormsProperties**

File: common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFile.java (hunk 1 above)
Comment (baptistemesta) [inline common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFile.java]: will be catched by the API layer and thrown as an API exception, is that ok?

**Comment by educhastenier** [comment]
AuthenticationManagerPropertiesFactory still looks for filr **authenticationManager-config.properties** on FS (in temp folder)

File: common/src/main/java/org/bonitasoft/console/common/server/auth/AuthenticationManagerProperties.java [hunk 3]
```diff
@@ -85,46 +80,23 @@
...
-                }
-            }
-        }
+
+    public AuthenticationManagerProperties(final long tenantId) {
+        super(AUTHENTICATION_CONFIG_FILE_NAME, tenantId >= 0 ? tenantId : TenantsManagementUtils.getDefaultTenantId());
+    }
+
+    public static AuthenticationManagerProperties getProperties(long tenanId) {
+        return new AuthenticationManagerProperties(tenanId);
```
Comment (educhastenier) [inline common/src/main/java/org/bonitasoft/console/common/server/auth/AuthenticationManagerProperties.java]: typo in parameter name

File: common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFilesManager.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+    private static ConfigurationFilesManager INSTANCE = new ConfigurationFilesManager();
+
+    public static ConfigurationFilesManager getInstance() {
+        return INSTANCE;
+    }
+
+    private Map<Long, Map<String, Properties>> tenantConfigurations = new HashMap<>();
+    private Map<String, Properties> platfomConfiguration = new HashMap<>();
+    private Map<String, File> platfomConfigurationFiles = new HashMap<>();
```
Comment (educhastenier) [inline common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFilesManager.java]: typo

File: common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFilesManager.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            }
+        }
+        tenantConfigurations.put(tenantId, tenantProperties);
+    }
+
+    public void removeProperty(String propertiesFilename, long tenantId, String propertyName) throws IOException {
+        //FIXME make it persisted in database
+        Map<String, Properties> resources = getResources(tenantId);
+        Properties properties = resources.get(propertiesFilename);
+        properties.remove(propertyName);
```
Comment (educhastenier) [inline common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFilesManager.java]: what if properties is null?

File: common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFilesManager.java [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+                tenantProperties.put(entry.getKey(), getProperties(entry.getValue()));
+            }
+        }
+        tenantConfigurations.put(tenantId, tenantProperties);
+    }
+
+    public void removeProperty(String propertiesFilename, long tenantId, String propertyName) throws IOException {
+        //FIXME make it persisted in database
+        Map<String, Properties> resources = getResources(tenantId);
+        Properties properties = resources.get(propertiesFilename);
```
Comment (educhastenier) [inline common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFilesManager.java]: what if resources is null?

File: common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFilesManager.java [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+        } else {
+            resources = platformConfiguration;
+        }
+        return resources;
+    }
+
+    public void setProperty(String propertiesFilename, long tenantId, String propertyName, String propertyValue) throws IOException {
+        //FIXME make it persisted in database
+        Map<String, Properties> resources = getResources(tenantId);
+        Properties properties = resources.get(propertiesFilename);
```
Comment (educhastenier) [inline common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFilesManager.java]: same remark

File: common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFilesManager.java [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+        properties.setProperty(propertyName, propertyValue);
+    }
+
+    public File getFile(String fileName) {
```
Comment (educhastenier) [inline common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFilesManager.java]: method name is not very explicit

File: common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/SecurityProperties.java [hunk 9]
```diff
@@ -211,13 +110,13 @@ private boolean isValidProcessDefinition(final String processDefinitionId) {
      *
      * @param webBonitaConstantsUtils
      * @param processDefinitionId
-     *            the process definition ID
+     *        the process definition ID
      * @return the config file or null if it doesn't exists
      * @throws IOException
      */
-    protected File getProcessSecurityPropertiesFile(final WebBonitaConstantsUtils webBonitaConstantsUtils, final String processDefinitionId) {
+    protected File getProcessSecurityPropertiesFile(final WebBonitaConstantsUtils webBonitaConstantsUtils, final ProcessIdentifier processDefinitionId) {
```
Comment (educhastenier) [inline common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/SecurityProperties.java]: wrong parameter name

File: common/src/main/java/org/bonitasoft/console/common/server/utils/PlatformManagementUtils.java [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+ * @author Baptiste Mesta
+ */
+public class PlatformManagementUtils {
+
+    private boolean isLocal() throws UnknownAPITypeException, ServerAPIException, IOException {
+        return ApiAccessType.LOCAL.equals(APITypeManager.getAPIType());
+    }
+
+    private PlatformSession platformLogin() throws BonitaException, IOException {
+        PlatformLoginAPI platformLoginAPI = PlatformAPIAccessor.getPlatformLoginAPI();
```
Comment (educhastenier) [inline common/src/main/java/org/bonitasoft/console/common/server/utils/PlatformManagementUtils.java]: put this declaration 8 lines lower

File: common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFilesManager.java (hunk 5 above)
Comment (baptistemesta) [inline common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFilesManager.java]: fixed in next PR

File: common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFilesManager.java (hunk 6 above)
Comment (baptistemesta) [inline common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFilesManager.java]: fixed in next PR

File: common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFilesManager.java (hunk 7 above)
Comment (baptistemesta) [inline common/src/main/java/org/bonitasoft/console/common/server/preferences/properties/ConfigurationFilesManager.java]: fixed in next PR

**Pull request merged**